src/main/java/Duke.java

```diff
@@ -1,10 +1,283 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+class DukeException extends Exception {
+    public DukeException(String message) {
+        super(message);
+    }
+}
+
 public class Duke {
+    private static final String FILE_PATH = "duke.txt";
+    private static ArrayList<Task> tasks;
+    public Duke() {
+        tasks = loadTasksFromFile();
+    }
+
     public static void main(String[] args) {
+        Duke duke = new Duke();
         String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm CharmBot ");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    
+        Scanner scanner = new Scanner(System.in);
+        String command = scanner.nextLine();
+         // Initialize tasks list here
+    
+        while (!command.equals("bye")) {
+            System.out.println("____________________________________________________________");
+            try {
+                if (command.equals("list")) {
```

Comment:
The code seems to be too long for me. Try to divide each commands into different functions.